keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -0,0 +1,1009 @@
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+the API proposed by this KEP aims to
+- Provide a reliable, consistent, and automated approach for any
+  multi-cluster application (framework, toolset) to discover available
+  clusters and take actions accordingly, in a way similar to service
+  discovery works in a microservice architecture. Through the inventory,
+  the application can query for a list of clusters to access, or watch
+  for an ever-flowing stream of cluster lifecycle events which the
+  application can act upon timely, such as auto-scaling, upgrades,
+  failures, and connectivity issues.
+- Provide a simple, clear interface for human operators to understand
+  clusters under management.
+
+### Terminology
+
+- **Cluster Manager**: An entity that creates the cluster inventory API 
+  object per member cluster, and keeps their status up-to-date. Each
+  cluster manager SHOULD be identified with a unique name. Each cluster
+  inventory resource SHOULD be managed by only one cluster manager. Examples
+  of cluster manager are projects like OCM, Karmada, Clusternet or Azure
+  fleet manager.
+
+- **Cluster Inventory Consumer**: the person running the cluster managers
+  or the person developing extensions for cluster managers for the purpose of
+  workload distribution, operation management etc.
+
+- **Member Cluster**: A kubernetes cluster that is managed by the cluster
+  manager. A cluster manager SHOULD have sufficient permission to access
+  the member cluster to fetch the information so it can update the status
+  of the cluster inventory API resource.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1: Multicluster Workload Distribution
+
+In this scenario, a standard API acts as a foundation for multicluster
+scheduling decisions across various clusters. This means that the
+API will provide a single point of contact to retrieve information that can be
+used to make informed scheduling decisions for workloads across multiple
+clusters. For instance, workload distribution tools like GitOps or
+[Work API](https://github.com/kubernetes-sigs/work-api) can leverage this
+API to make informed decisions about which cluster is best suited to handle
+a particular workload.
+
+Examples of the multicluster scheduling includes:
+
+- As a user I want to run a workload on an EKS cluster that resides in us-east-1.
+  I want to submit my workload if and only if a cluster satisfies that constraint.
+- As a user I want to run a workload on a cluster that has certain CRDs installed.
+- As a user I want to deploy a workload to my on-prem cluster group.
```

Comment:
cluster group means something different to each reader, as we're not implying grouping in this KEP, it might be better to just reference "...to my on-prem cluster."